packages/components/pie-modal/src/index.ts

```diff
@@ -144,55 +159,47 @@ export class PieModal extends RtlMixin(LitElement) {
     }
 
     /**
-     * Dismisses the modal on backdrop click
+     * Dismisses the modal on backdrop click. (dependent on prop: `isDismissible`)
      *
      */
     private _handleDialogLightDismiss = (event: MouseEvent) : void => {
-        const rect = this._dialog?.getBoundingClientRect();
+        if (this.isDismissible) {
+            const rect = this._dialog?.getBoundingClientRect();
 
-        const {
-            top = 0, bottom = 0, left = 0, right = 0,
-        } = rect || {};
+            const {
+                top = 0, bottom = 0, left = 0, right = 0,
+            } = rect || {};
 
-        // This means the dialog is not open due to clicking the close button so
-        // we want to escape early
-        if (top === 0 && bottom === 0 && left === 0 && right === 0) {
-            return;
-        }
+            // This means the dialog is not open due to clicking the close button so
+            // we want to escape early
+            if (top === 0 && bottom === 0 && left === 0 && right === 0) {
+                return;
+            }
 
-        const isClickOutsideDialog = event.clientY < top ||
-            event.clientY > bottom ||
-            event.clientX < left ||
-            event.clientX > right;
+            const isClickOutsideDialog = event.clientY < top ||
+                event.clientY > bottom ||
+                event.clientX < left ||
+                event.clientX > right;
 
-        if (isClickOutsideDialog) {
-            this.isOpen = false;
+            if (isClickOutsideDialog) {
+                this.isOpen = false;
+            }
         }
     };
 
     /**
-     * Dispatch `ON_MODAL_CLOSE_EVENT` event.
-     * To be used whenever we close the modal.
+     * Note: We should aim to have a shareable event helper system to allow
+     * us to share this across components in-future.
      *
-     * @event
-     */
-    private _dispatchModalCloseEvent = () : void => {
-        const event = new CustomEvent(ON_MODAL_CLOSE_EVENT, {
-            bubbles: true,
-            composed: true,
-        });
-
-        this.dispatchEvent(event);
-    };
-
-    /**
-     * Dispatch `ON_MODAL_OPEN_EVENT` event.
-     * To be used whenever we open the modal.
+     * Dispatch a custom event.
+     *
+     * To be used whenever we have behavioural events we want to
+     * bubble up through the modal.
      *
-     * @event
+     * @eventType
```

Comment:
does this comment need some additional information?

Reply:
Do you mean the overall method or just a comment about the eventType? I have a ticket around this which i will fill out soon, that ticket is to over our event system across components.

Reply:
The event type jsdoc comment :)

Reply:
fixed.